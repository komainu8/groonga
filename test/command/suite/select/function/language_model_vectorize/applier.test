#@omit Need language mode management feature

#@require-feature llama.cpp

plugin_register functions/language_model
plugin_register functions/vector

table_create Data TABLE_NO_KEY
column_create Data text COLUMN_SCALAR ShortText

load --table Data
[
{"text": "I am a king."},
{"text": "I am a queen."}
]

#@timeout 60
#@read-timeout 30
select Data \
  --columns[embeddings].stage output \
  --columns[embeddings].type Float32 \
  --columns[embeddings].flags COLUMN_VECTOR \
  --columns[embeddings].value 'language_model_vectorize("mistral-7b-v0.1.Q4_K_M", text)' \
  --output_columns 'text, vector_slice(embeddings, 0, 3)'
