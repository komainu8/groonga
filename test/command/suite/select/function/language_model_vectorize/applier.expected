plugin_register functions/language_model
[[0,0.0,0.0],true]
plugin_register functions/vector
[[0,0.0,0.0],true]
table_create Data TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Data text COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
load --table Data
[
{"text": "I am a king."},
{"text": "I am a queen."}
]
[[0,0.0,0.0],2]
select Data   --columns[embeddings].stage output   --columns[embeddings].type Float32   --columns[embeddings].flags COLUMN_VECTOR   --columns[embeddings].value 'language_model_vectorize("mistral-7b-v0.1.Q4_K_M", text)'   --output_columns 'text, vector_slice(embeddings, 0, 3)'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "text",
          "ShortText"
        ],
        [
          "vector_slice",
          null
        ]
      ],
      [
        "I am a king.",
        [
          0.008547269,
          0.006048318,
          0.003951299
        ]
      ],
      [
        "I am a queen.",
        [
          0.007824666,
          0.00421184,
          0.00569254
        ]
      ]
    ]
  ]
]
#|w| llm_load_vocab: special_eos_id is not in special_eog_ids - the tokenizer config may be incorrect
